File: vehicle-service/internal/infrastructure/api/dtos/vehicle.go
package dtos

import "time"

// CreateVehicleRequest is the request body for POST /api/vehicles.
type CreateVehicleRequest struct {
	Plate string `json:"plate"`
}

// VehicleResponse is the response body for single-vehicle endpoints.
type VehicleResponse struct {
	ID        int64     `json:"id"`
	Plate     string    `json:"plate"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}

// ListVehiclesResponse is the paginated response for GET /api/vehicles.
type ListVehiclesResponse struct {
	Data   []VehicleResponse `json:"data"`
	Limit  int               `json:"limit"`
	Offset int               `json:"offset"`
	Total  int64             `json:"total"`
}

File: telemetry-movil/src/api/vehicle.ts
import { request } from "./client";
import type { Vehicle } from "./types";

// createVehicle calls POST /api/vehicles with the given plate.
export function createVehicle(plate: string): Promise<Vehicle> {
  return request<Vehicle>("/api/vehicles", {
    method: "POST",
    body: JSON.stringify({ plate }),
  });
}

// findByPlate calls GET /api/vehicles/plates/{plate}.
export function findByPlate(plate: string): Promise<Vehicle> {
  return request<Vehicle>(`/api/vehicles/plates/${encodeURIComponent(plate)}`, {
    method: "GET",
  });
}
